lib/smart_proxy_dns_powerdns/backend/mysql.rb

```diff
@@ -35,15 +35,20 @@ def create_record domain_id, name, type, content
       name = connection.escape(name)
       content = connection.escape(content)
       type = connection.escape(type)
-      connection.query("INSERT INTO records (domain_id, name, ttl, content, type) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}')")
+      connection.query("INSERT INTO records (domain_id, name, ttl, content, type, change_date) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}', UNIX_TIMESTAMP())")
       connection.affected_rows == 1
     end
 
     def delete_record domain_id, name, type
       name = connection.escape(name)
       type = connection.escape(type)
       connection.query("DELETE FROM records WHERE domain_id=#{domain_id} AND name='#{name}' AND type='#{type}'")
-      connection.affected_rows >= 1
+      ret = connection.affected_rows >= 1
+      if ret
+        connection.query("UPDATE records SET change_date=UNIX_TIMESTAMP() WHERE domain_id=#{domain_id} AND type='SOA'")
+        ret = connection.affected_rows == 1
```

Comment:
> Should I discard the return value of the UPDATE statement then?
> Maybe, at least issuing a warning would be better than silently ignoring the result.

I think it should be an info message. The race condition is quite possible and I'd rather not have false positive warnings. The other cases aren't that likely and I suspect you'll notice the breakage in other ways. If there is no SOA then PowerDNS won't serve the zone and no idea what happens when you have multiple SOA records. Maybe that should be a warning.